platforms/core-configuration/declarative-dsl-provider/src/main/kotlin/org/gradle/internal/declarativedsl/project/dependencyConfigurationSchema.kt

```diff
@@ -103,6 +133,48 @@ class RuntimeDependencyFunctionResolver(configurations: DependencyConfigurations
                 }
             })
         }
+
         return RuntimeFunctionResolver.Resolution.Unresolved
     }
 }
+
+
+private
+class ImplicitDependencyCollectorFunctionResolver(configurations: DependencyConfigurations) : RuntimeFunctionResolver {
+    private
+    val configurationNames = configurations.configurationNames.toSet()
+
+    override fun resolve(receiverClass: KClass<*>, name: String, parameterValueBinding: ParameterValueBinding): RuntimeFunctionResolver.Resolution {
+        if (name in configurationNames) {
+            val getterFunction = getDependencyCollectorGetter(receiverClass, name)
```

Comment:
💭 An alternative to consider would be to delegate getting the dependency collector to a `ReflectionRuntimePropertyResolver` (especially if you decide to support Kotlin properties as well, see the other comment) and then just invoke `add` on the result.